### PR TITLE
store: fix data race in mock PD timeout error

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve? 
Fix data race in mock for PD timeout error.

### What is changed and how it works?
Fix it.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

### CODE REVIEW

File: session/session_fail_test.go [hunk 1]
```diff
@@ -102,7 +102,8 @@ func (s *testSessionSuite) TestGetTSFailDirtyStateInretry(c *C) {
 	tk.MustExec("create table t (id int)")
 
 	c.Assert(failpoint.Enable("github.com/pingcap/tidb/session/mockCommitError", `return(true)`), IsNil)
-	c.Assert(failpoint.Enable("github.com/pingcap/tidb/store/tikv/mockGetTSErrorInRetry", `return(true)`), IsNil)
+	c.Assert(failpoint.Enable("github.com/pingcap/tidb/store/tikv/mockGetTSErrorInRetry",
+		`1*return(true)->return(false)`), IsNil)
```
Comment (jackysp) [inline session/session_fail_test.go]: This modification is according to 

https://github.com/pingcap/failpoint/blob/53525bdd7953438b0255d33962bc1fe6b1d57727/terms_test.go#L43

Comment (ngaut) [reply session/session_fail_test.go]: Is this mean return false just once? Could you add comments to explain what is the purpose of 1?

Comment (jackysp) [reply session/session_fail_test.go]: It means return true once then all falses. This case wants to mock PD timeout error could recover automatically by the retry mechanism. I'll add some comments.

**Pull request merged**